docs/src/modules/components/AppSettingsDrawer.js

```diff
@@ -36,6 +36,72 @@ const IconToggleButton = styled(ToggleButton)({
   },
 });
 
+function ToggleTheme(props) {
+  const { value, onChange } = props;
+  const t = useTranslate();
+  return (
+    <ToggleButtonGroup
+      exclusive
+      value={value}
+      onChange={onChange}
+      color="primary"
+      aria-labelledby="settings-mode"
+      fullWidth
+    >
+      <IconToggleButton
+        value="light"
+        aria-label={t('settings.light')}
+        data-ga-event-category="settings"
+        data-ga-event-action="light"
+      >
+        <LightModeIcon fontSize="small" />
+        {t('settings.light')}
+      </IconToggleButton>
+      <IconToggleButton
+        value="system"
+        aria-label={t('settings.system')}
+        data-ga-event-category="settings"
+        data-ga-event-action="system"
+      >
+        <SettingsBrightnessIcon fontSize="small" />
+        {t('settings.system')}
+      </IconToggleButton>
+      <IconToggleButton
+        value="dark"
+        aria-label={t('settings.dark')}
+        data-ga-event-category="settings"
+        data-ga-event-action="dark"
+      >
+        <DarkModeOutlinedIcon fontSize="small" />
+        {t('settings.dark')}
+      </IconToggleButton>
+    </ToggleButtonGroup>
+  );
+}
+
+function ToggleVarTheme(props) {
+  const { setMode } = useColorScheme();
```

Comment:
The hook requires to be in a context of a theme with vars otherwise it throws error.
That's the reason why `ToggleTheme` is split into 2 components:

- On for the CSS var if feasible
- ANother one without `useColorScheme();` if `theme.vars` is not detected